pkg/sensors/tracing/generickprobe.go

```diff
@@ -1080,6 +1112,37 @@ func handleGenericKprobe(r *bytes.Reader) ([]observer.Event, error) {
 		printers = gk.argSigPrinters
 	}
 
+	if m.Common.Flags&processapi.MSG_COMMON_FLAG_STACKTRACE != 0 {
+		if m.StackID < 0 {
+			logger.GetLogger().Warnf("failed to retrieve stacktrace: id equal to errno %d", m.StackID)
+		} else {
+			// remove the error part
+			id := uint32(m.StackID)
+
+			// lazy load the map reference if needed
+			if gk.stackTraceMapRef == nil {
+				bpf.MapPrefixPath()
+				gk.stackTraceMapRef, err = ebpf.LoadPinnedMap(path.Join(bpf.MapPrefixPath(), gk.pinPathPrefix)+"-stack_trace_map", &ebpf.LoadPinOptions{
+					ReadOnly: true,
+				})
+				if err != nil {
+					logger.GetLogger().WithError(err).Warn("failed to load the stacktrace map")
+				}
+				// close this in cleanup postHook defer stackTraceMap.Close()
+			}
+
+			// this can't be an else statement in the previous block since it
+			// must execute as well when the reference is first initialized
+			if gk.stackTraceMapRef != nil {
```

Comment:
But can it be that LoadPinnedMap returns a not-nil value, without the error?
In other words, do we need the `if` statement? Isn't it always true?

Reply:
with the actual code, we need to check since `err != nil` doesn't prevent us to continue to execute since we just log the error.

Reply:
with the actual code, we need to check since `err != nil` doesn't prevent us to continue to execute since we just log the error.

Reply:
I see. I'm fine with keeping as is. Another option might be to move the code into a separate function from which we can just `return` if there is an error to simplify the flow.